libfreerdp/common/settings.c

```diff
@@ -2432,231 +2445,232 @@ char* freerdp_get_param_string(rdpSettings* settings, int id)
 
 int freerdp_set_param_string(rdpSettings* settings, int id, const char* param)
 {
+#define CHECKED_STRDUP(name) if (param && !(settings->name = _strdup(param))) return -1
```

Comment:
IMHO the solution you've used in freerdp_client_rdp_file_set_string() with the tmp LPSTR* variable is more elegant.